lib/thread.h

```diff
@@ -43,6 +43,8 @@ struct rusage_t {
 PREDECL_LIST(thread_list)
 PREDECL_HEAP(thread_timer_list)
 
+struct vty;
```

Comment:
Why do we need to change lib/thread.h?  no-one is going to be using this but thread.c.  Make these functions static

Reply:
Have added new APIs thread_dump_master_statistics(struct vty *vty, struct thread_master *m) and thread_dump_all_thread_statistics(struct vty *vty) for any other component specific show to utilise and include the information in the component-specific command output. 

We could have '#include <vty.h>" here. But that would have cuased a recursive inclusion loop. Hence chose to forward declare struct vty to avoid the loop.

Reply:
No other compoents are using it.  So no.

Reply:
Okay will remove it. But to remove it I will also have to make the new thread_dump_xxxx() functions static from global. Hope that is fine. Please confirm.